Weno new rx standard 3

juggernautsei opened the pull request:
<!--
(Thanks for sending a pull request!
-->

<!-- Add the issue number that is fixed by this PR (In the form Fixes #123) -->
Fixes #

#### Short description of what this resolves:
The changes requested for the PR

#### Changes proposed in this pull request:

- encodeURIComponent() added
- move an ajax call to a class file
-

bradymiller commented:
hi @juggernautsei ,
I just left comments in the prior PR (https://github.com/openemr/openemr/pull/3188). We will continue to now work from this new PR.

juggernautsei commented:
We have some more changes coming. I have third party confirmation that the prescriptions can go all the way through the gateway to be signed at Weno's server.
There will be a couple of more pieces of information that will be added to the user form. We need to get the prescriber's email address to send with the script. There needs to be a sig in addition to the note box on the prescription form. We are working those things out and will post as soon as we can. Dr. Hausam is assisting in the code build.

stephenwaite commented on sql/5_0_1-to-5_0_2_upgrade.sql:
```diff
@@ -705,6 +705,10 @@ ALTER TABLE `facility` ADD `mail_zip` VARCHAR(10) default NULL;
+ALTER TABLE `facility` ADD `weno_id` int(e) default NULL;
+#EndIf
+
 #IfNotTable keys
```
hi @juggernautsei , only need to add this in the 502 -> 503 script below

juggernautsei replied:
Removed

stephenwaite commented on interface/usergroup/facility_admin.php:
```diff
@@ -201,6 +201,14 @@ function displayAlert() {
+                            <label for="weno_id" class='col-form-label col-form-label-sm'><?php echo xlt('Weno ID'); ?>:</label>
+                        </div>
+                        <div class="col">
+                            <input type="entry" class="form-controlform-control-sm" size="20" name="weno_id" value="<?php echo attr($facility["weno_id"]); ?>">
```
looks like a typo, should be `form-control form-control-sm`

juggernautsei replied:
@stephenwaite I could not figure out how to get the weno id to save. I know the data is being serialized and that where I am lost. I know it is being sent via ajax but could not track down where it is being unserialized. Do you know? Thanks!

juggernautsei replied:
I figured it out. My dev was broken.

juggernautsei commented:
The system has been tested in a production environment by Dr. Robert Huasem.
I have updated the patch and added a new video to explain the facility ID(s) add.  
https://www.open-emr.org/wiki/index.php/OpenEMR_ePrescribe#JSeRx

juggernautsei commented:
@stephenwaite I am ready to pick this up again. I was trying to rebase the commit. But ran into the same issue as above Conflicts. I will be doing some code cleanup on this commit as well. Should I close this branch and create a new branch and PR?
![image](https://user-images.githubusercontent.com/834958/86009292-0bd0b580-b9e8-11ea-82d6-cd7a1ca45e2a.png)

stephenwaite commented:
hi @juggernautsei , sounds good, thank you
